### PR TITLE
add workflows for PRs

### PR DESCRIPTION
**Pull request by mtar**
## Description

Adds workflows that get activated when a new PR is created. Runs a unit test on a single process and returns a message on success. The local PR branch is then send to the mirrored repository.

Pushes on the main and release branches are mirrored only to avoid double testing of branches/PR on the upstream repository.

Issue/s resolved: #1113 

## Changes proposed:
- add workflows for PRs
- update existing workflow for mirroring

## Type of change

- workflow update

### CODE REVIEW

File: .github/workflows/ReceivePR.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Receive PR
+
+on:
+  pull_request:
+
```
Comment (ClaudiaComito) [inline .github/workflows/ReceivePR.yml:5]: Shall we add a `path-ignore:` filter for workflows and docs?

File: .github/workflows/ci_cb.yml [hunk 2]
```diff
@@ -1,6 +1,10 @@
-name: Mirror and run GitLab CI
+name: CI run
 
-on: [push]
+on:
+  push:
+    branches:
+      - 'main'
+      - 'release/**'
```
Comment (ClaudiaComito) [inline .github/workflows/ci_cb.yml:7]: OK, so after this PR is merged, tests will not run after every push on every branch like we're used to. They'll only run once a PR is created. Is that correct?

**Review by ClaudiaComito — COMMENTED** [review]
Thanks @mtar , it looks good to me, I only have minor comments but as far as I'm concerned we can merge this.

File: .github/workflows/ci_cb.yml (hunk 2 above)
Comment (mtar) [reply .github/workflows/ci_cb.yml:7]: Yes, I avoid triggering the pipeline twice.

File: .github/workflows/ReceivePR.yml (hunk 1 above)
Comment (mtar) [reply .github/workflows/ReceivePR.yml:5]: I can do it. We will have to bypass the branch protection then.

Comment (ClaudiaComito) [reply .github/workflows/ReceivePR.yml:5]: ok then let's not.

Comment (mtar) [reply .github/workflows/ReceivePR.yml:5]: I don't have any issues with that. These kind of PRs are not common and don't have any impact on the tests.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ my be illegible. It may be easiest to save the output of each to a file.
 - [ ] All split configurations tested
 - [ ] Multiple dtypes tested in relevant functions
 - [ ] Documentation updated (if needed)
-- [ ] Title of PR is suitable for corresponding CHANGELOG entry
+- [ ] Updated changelog.md under the title "Pending Additions"
```
Comment (ClaudiaComito) [inline .github/PULL_REQUEST_TEMPLATE.md:49]:
```suggestion
- [ ] Title of PR is suitable for corresponding CHANGELOG entry
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -30,13 +30,13 @@ jobs:
     name: Python ${{ matrix.py-version }} with ${{ matrix.pytorch-version }}; options ${{ matrix.install-options }}
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v2
```
Comment (ClaudiaComito) [inline .github/workflows/ci.yaml:33]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -36,7 +36,7 @@ jobs:
         with:
           mpi: ${{ matrix.mpi }}
       - name: Use Python ${{ matrix.py-version }}
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v2
```
Comment (ClaudiaComito) [inline .github/workflows/ci.yaml:39]:
```suggestion
        uses: actions/setup-python@v4
```

File: heat/core/_operations.py [hunk 6]
```diff
@@ -1,4 +1,6 @@
-"""Generalized operations for DNDarray"""
+"""
+Generalized MPI operations. i.e. element-wise binary operations
+"""
```
Comment (ClaudiaComito) [inline heat/core/_operations.py:3]:
```suggestion
"""
Generalized operations for DNDarray
"""
```

**Review by ClaudiaComito — DISMISSED** [review]

**Review by ClaudiaComito — APPROVED** [review]
Merging to `main`

**Pull request merged**